packages/spanner/src/database.js

```diff
@@ -400,6 +400,64 @@ Database.prototype.getSchema = function(callback) {
 };
 
 /**
+ * Get a read/write ready Transaction object.
+ *
+ * @param {object=} options - Transaction options.
+ * @param {boolean} options.readOnly - Specifies if the transaction is read
+ *     only.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while getting the
+ *     transaction object.
+ * @param {module:spanner/transaction} transaction - The transaction object.
+ *
+ * @example
+ * database.getTransaction(function(err, transaction) {});
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * database.getTransaction().then(function(data) {
+ *   var transaction = data[0];
+ * });
+ */
+Database.prototype.getTransaction = function(options, callback) {
+  var self = this;
+
+  if (is.fn(options)) {
+    callback = options;
+    options = null;
+  }
+
+  if (!options || !options.readOnly) {
+    this.pool_.getWriteSession(function(err, session, transaction) {
+      callback(err, transaction);
+    });
+    return;
+  }
+
+  this.pool_.getSession(function(err, session) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    options = extend({}, options);
+    delete options.readOnly;
+
+    var transaction = self.pool_.createTransaction_(session, options);
+
+    transaction.begin(function(err) {
+      if (err) {
+        callback(err);
```

Comment:
But in this error case since the transaction is not even passed to the user specified callback, how would they end it?